tests/expected/cpp/60164-Issue_3191.cpp

```diff
@@ -0,0 +1 @@
+if (foo) return {};
```

Comment:
I don't understand. The test config is asking for a newline to be added after an `if` but there is newline here. This seems not correct. Any idea?

Reply:
I understand that the original code has a bug, but the fix does not seem to be correct because the result does not behave as would be expected given the provided config.
I think we need a better solution, that fix the bug and behaves consistenly

Reply:
`if (true) {...}<here>` yes, that is where it is supposed to add the newline. If `nl_after_if = add` uncrustify make sure there is at least one new line after the `if` block. Keep in mind that the end of the block is `}` and not `};`. Technically the `;` is a second statement.
The new test code is better, the original one was not showing any newline after the `if` statement.
Now the question is "should the newline be after `}` or after `};`? This can be a very subjective point of view. From one point of view, `};` looks better. From another point of view, having `}` followed by newline and a spare `;` would help identifying mistakes in a quicker way.
@pavelgeiger What do you think?
@guy-maurel would also like to hear your opinion.

Reply:
I think newline be after } is the better option

Reply:
I also think newline after `}` is most appropriate.
Looking at the original reported example, it seems two new lines are added rather than one, so that is something that should be address. The result should be something like this:
```
if (foo)
    return {}
;
```
which allows to clearly spot unnecessary semicolons. And the option `mod_remove_extra_semicolon` allows to remove them automatically if needed.